hardware/esp8266com/esp8266/boards.txt

```diff
@@ -57,6 +58,55 @@ generic.menu.FlashSize.4M=4M
 generic.menu.FlashSize.4M.build.flash_size=4M
 
 ##############################################################
+modwifi.name=Olimex MOD-WIFI-ESP8266(-DEV)
+
+modwifi.upload.tool=esptool
+modwifi.upload.speed=115200
+modwifi.upload.resetmethod=ck
+modwifi.upload.maximum_size=2097152
+modwifi.upload.wait_for_upload_port=true
+modwifi.serial.disableDTR=true
+modwifi.serial.disableRTS=true
+
+modwifi.build.mcu=esp8266
+modwifi.build.f_cpu=80000000L
+modwifi.build.board=MOD_WIFI_ESP8266
+modwifi.build.core=esp8266
+modwifi.build.variant=modwifi
+# Winbond W25Q16 flash
+modwifi.build.flash_mode=qio
+modwifi.build.flash_size=2M
+modwifi.build.flash_freq=40
+
+modwifi.menu.CpuFrequency.80=80 MHz
+modwifi.menu.CpuFrequency.80.build.f_cpu=80000000L
+modwifi.menu.CpuFrequency.160=160 MHz
+modwifi.menu.CpuFrequency.160.build.f_cpu=160000000L
+
+modwifi.menu.UploadSpeed.115200=115200
+modwifi.menu.UploadSpeed.115200.upload.speed=115200
+modwifi.menu.UploadSpeed.9600=9600
+modwifi.menu.UploadSpeed.9600.upload.speed=9600
+modwifi.menu.UploadSpeed.57600=57600
+modwifi.menu.UploadSpeed.57600.upload.speed=57600
+modwifi.menu.UploadSpeed.256000.windows=256000
+modwifi.menu.UploadSpeed.256000.upload.speed=256000
+modwifi.menu.UploadSpeed.230400.linux=230400
+modwifi.menu.UploadSpeed.230400.macosx=230400
+modwifi.menu.UploadSpeed.230400.macosx=230400
+modwifi.menu.UploadSpeed.230400.upload.speed=230400
+modwifi.menu.UploadSpeed.460800.linux=460800
+modwifi.menu.UploadSpeed.460800.macosx=460800
+modwifi.menu.UploadSpeed.460800.upload.speed=460800
+modwifi.menu.UploadSpeed.512000.windows=512000
+modwifi.menu.UploadSpeed.512000.upload.speed=512000
+modwifi.menu.UploadSpeed.921600=921600
+modwifi.menu.UploadSpeed.921600.upload.speed=921600
+
+modwifi.menu.FlashSize.2M=2M
```

Comment:
If 2M is the only option for flash size, you can remove this. No need to have a menu with just a single option.